packages/block-library/src/embed/youtube-controls.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { PanelBody, ToggleControl, TextControl } from '@wordpress/components';
+import { InspectorControls } from '@wordpress/editor';
+
+export const YouTubeInspectorControls = class extends Component {
+	constructor() {
+		super( ...arguments );
+		const { extraOptions = {} } = this.props.attributes;
+		this.setAttributes = this.setAttributes.bind( this );
+		this.commitAttributes = debounce( this.props.setAttributes, 1000 );
+		this.state = {
+			autoplay: extraOptions.autoplay || false,
+			start: extraOptions.start || 0,
+			relatedOnlyFromChannel: extraOptions.relatedOnlyFromChannel || false,
+		};
+	}
+
+	setAttributes( attributes ) {
+		const { extraOptions = {} } = this.props.attributes;
+		this.setState( attributes );
+		this.commitAttributes( { extraOptions: { ...extraOptions, ...attributes } } );
+	}
+
+	render() {
+		const { autoplay, start, relatedOnlyFromChannel } = this.state;
+		const onChangeAutoplay = ( value ) => {
+			this.setAttributes( { autoplay: value } );
+		};
+		const onChangeRelated = ( value ) => {
+			this.setAttributes( { relatedOnlyFromChannel: value } );
+		};
+		const onChangeStart = ( value ) => {
+			this.setAttributes( { start: value } );
+		};
```

Comment:
Minor note but we are creating new functions on each render call. The value is being passed from the event, so we can pull this out of render. Were you having problems with `this`?

Reply:
Ugh, no real reason other than took the shortest route to making it work so I could get feedback on the approach :) definitely should be moved out